extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java

```diff
@@ -478,14 +445,24 @@ void registerHealthUiHandler(
             BuildProducer<RouteBuildItem> routeProducer,
             SmallRyeHealthRecorder recorder,
             SmallRyeHealthRuntimeConfig runtimeConfig,
-            SmallRyeHealthBuildItem smallRyeHealthBuildItem,
+            WebJarResultBuildItem webJarResultBuildItem,
             NonApplicationRootPathBuildItem nonApplicationRootPathBuildItem,
             LaunchModeBuildItem launchMode,
-            SmallRyeHealthConfig healthConfig) {
+            SmallRyeHealthConfig healthConfig,
+            BuildProducer<SmallRyeHealthBuildItem> smallryeHealthBuildProducer) {
+
+        WebJarResultBuildItem.WebJarResult result = webJarResultBuildItem.byArtifactKey(HEALTH_UI_WEBJAR_ARTIFACT_KEY);
+        if (result == null) {
+            return;
+        }
 
         if (shouldInclude(launchMode, healthConfig)) {
-            Handler<RoutingContext> handler = recorder.uiHandler(smallRyeHealthBuildItem.getHealthUiFinalDestination(),
-                    smallRyeHealthBuildItem.getHealthUiPath(), runtimeConfig);
+            String healthUiPath = nonApplicationRootPathBuildItem.resolvePath(healthConfig.ui.rootPath);
+            smallryeHealthBuildProducer
+                    .produce(new SmallRyeHealthBuildItem(result.getFinalDestination(), healthUiPath));
```

Comment:
`SmallRyeHealthBuildItem` just to be on the safe side, no idea if anyone consumes it.

Maybe we can deprecate it also?